File: src/main/java/de/chaos/swlnmngr/route/Router.java
package de.chaos.swlnmngr.route;

import de.chaos.swlnmngr.Main;
import de.chaos.swlnmngr.route.routes.*;

import java.util.*;

public class Router {

    private static final List<Route> ROUTES;

    static {
        ROUTES = List.of(new InstallRoute(), new UpdateRoute(), new LinkRoute(), new NewRoute());
    }

    public static void route(String[] args) {
        for (Route route : ROUTES) {
            if(route.getName().equalsIgnoreCase(args[0])) {
                String[] rArgs = new String[args.length - 1];
                System.arraycopy(args, 1, rArgs, 0, args.length - 1);
                Main.getLogger().info("Running: {}", route.getName());
                if(route.route(rArgs)) {
                    System.exit(0);
                } else {
                    System.exit(1);
                }
            }
        }
    }

    public static void printRoutes() {
        Main.getLogger().info("Available Routes: ");
        Main.getLogger().info("\t{}", ROUTES.stream().map(Route::getName).reduce((s, s2) -> s + ", " + s2).get());
    }
}

File: src/main/java/de/chaos/swlnmngr/route/Router.kt
package de.chaos.swlnmngr.route

import de.chaos.swlnmngr.logger
import de.chaos.swlnmngr.route.routes.*
import org.apache.logging.log4j.Level
import kotlin.system.exitProcess

val ROUTES :List<Route> = listOf(InstallRoute(), UpdateRoute(), LinkRoute(), NewRoute(), ReplRoute(), UpdateJarRoute())

fun route(args: Array<String>): Boolean {
    for (route in ROUTES) {
        if(route.getName().lowercase() == args[0].lowercase()) {
            val rArgs :Array<String> = Array(args.size - 1) { i: Int -> args[i + 1] }
            logger.info("Running: {}", route.getName())
            return route.route(rArgs)
        }
    }
    return false;
}

fun printRoutes() {
    logger.log(Level.INFO, "Available Routes: ")
    logger.log(Level.INFO, "\t{}", ROUTES.map { route -> route.getName() }.reduce { acc, s -> "$acc, $s" })
}